tools/protodoc/protodoc.py

```diff
@@ -92,6 +105,15 @@
         'This extension is work-in-progress. Functionality is incomplete and it is not intended for production use.',
 }
 
+EXTENSION_DB = json.loads(pathlib.Path(os.getenv('EXTENSION_DB_PATH')).read_text())
+
+# create an index of extension categories from extension db
+EXTENSION_CATEGORIES = {}
+for _k, _v in EXTENSION_DB.items():
+  for _cat in _v['categories']:
+    EXTENSION_CATEGORIES.setdefault(_cat, [])
+    EXTENSION_CATEGORIES[_cat].append(_k)
```

Comment:
@htuch i removed the script and shifted the code here

this simplifies/reduces the code and also parses the extension db only once, which should improve the efficiency of protodoc

i have used setdefault as `defaultdict` adds some complexity and would prevent `KeyErrors` from being raised below